pypeclub/OpenPype
# Deadline: reworked pools assignment

## Brief description
Previous solution was nonconsistent, expected values for pools in multiple places in Setting.

## Description
New collector and validator was introduced. Collector adds pool names from Setting (merged and removed separate pool settings in hosts sections) if not already present on a instance (as Maya has it, potentially others with New Publisher).

Validator checks if name of pools on the instances matches to current values on DL (does a query of DL).

## Additional info
Refactored `_requests_get` and `requests_post` wrappers into `openpype.lib.connections` - name up for commenting/changing.


## Testing notes:
1. configure `project_settings/deadline/publish/CollectDeadlinePools`
1.a - insert non existent values
1.b - insert values matching situation on DL
3. publish on DL and check pools
